hello-security/signup-controller.php

```diff
@@ -0,0 +1,31 @@
+<?php
+include "db-connection.php";
+
+if (strcmp($_POST["role"], "teacher") == 0 || 
+	strcmp($_POST["role"], "student") == 0) {
```

Comment:
Maybe it would be better to compare if the role is equal to"admin". So if you add a new role you won't need to change the code.

Reply:
But by doing this, user can sign up with a role that doesn't exist at all. 
Also, there is a small number of possible roles, so the extension of code won't be too large if any change is needed.

Reply:
Good point. In this case it is better to keep it like it is. Sometimes there is a second level of authorization management that is created - roles are related to a specific group and this way you can control the whole group. E.g. group `university_members` can contain `student` and `teacher` roles. But this is just fyi - you just may be curious to look at it further.